### PR TITLE
Issue 689: preview PR

### PR DESCRIPTION
**Pull request by lorban**
There's some work left, mostly in the area of testing. But comments already are welcome.

### CODE REVIEW

**Comment by lorban** [comment]
test this please

**Comment by lorban** [comment]
test this please

File: impl/src/main/java/org/ehcache/config/executor/PooledExecutionServiceConfiguration.java [hunk 1]
```diff
@@ -42,8 +42,11 @@ public void addDefaultPool(String alias, int minSize, int maxSize) {
   }
 
   public void addPool(String alias, int minSize, int maxSize) {
+    if (alias == null) {
+      throw new IllegalArgumentException("Pool alias cannot be null");
```
Comment (ljacomet) [inline impl/src/main/java/org/ehcache/config/executor/PooledExecutionServiceConfiguration.java]: NPE?

**Comment by ljacomet** [comment]
test this please

File: impl/src/main/java/org/ehcache/config/executor/PooledExecutionServiceConfiguration.java (hunk 1 above)
Comment (lorban) [inline impl/src/main/java/org/ehcache/config/executor/PooledExecutionServiceConfiguration.java]: Agreed. Let me change that.

**Pull request closed without merging**